app/serializers/shop_advertisement_serializer_mixin.rb

```diff
@@ -12,11 +12,15 @@ def shop_advertisement_html
 
     target_user_id = best_answer_id ? Post.find(best_answer_id).user_id : user_id
 
-    shop_id = UserCustomField.where(user_id: target_user_id, name: 'storeId').first&.value ||
-              Discourse.cache.read(:store_ids_for_advertisement)&.sample
+    if User.find(user_id).groups.where(name: SiteSetting.DiscourseDesignBundles_trusted_group).count > 0
```

Comment:
1.
We probably can go with `user` association without additional selection `User.find`. Check the topic whether user is mandatory for it

2.
Try to find a better way for `.count > 0`
Use something like `.any?` `.exists?`